### PR TITLE
Add In memory queue + tests

### PR DESCRIPTION
**Pull request by yotamloe**

### CODE REVIEW

File: inMemoryQueue/inMemoryQueue.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		return item, nil
+	}
+	fmt.Printf("Queue is full dropping logs\n")
+	return nil, nil
+}
+
+type Item = queue.Item
+
+func (c *ConcurrentQueue) Dequeue() (*Item, error) {
+	for c.isEmpty() {
```
Comment (Doron-Bargo) [inline inMemoryQueue/inMemoryQueue.go]: why we need an error if the queue is empty?

File: logziosender.go [hunk 2]
```diff
@@ -79,11 +86,14 @@ func New(token string, options ...SenderOptionFunc) (*LogzioSender, error) {
...
+		// In memory queue
+		inMemoryQueue:    false,
+		inMemoryCapacity: defaultQueueMaxLength,
+		logCountLimit:    defaultMaxLogCount,
 	}
-
 	tlsConfig := &tls.Config{}
 	transport := &http.Transport{
-		Proxy: http.ProxyFromEnvironment,
+		Proxy:           http.ProxyFromEnvironment,
```
Comment (Doron-Bargo) [inline logziosender.go:99]: please fix linting

File: logziosender.go [hunk 3]
```diff
@@ -262,24 +315,33 @@ func (l *LogzioSender) Drain() {
...
+					reDrain = true
+					break
+				}
 			}
 		}
 	}
+
 }
 
 func (l *LogzioSender) dequeueUpToMaxBatchSize() int {
```
Comment (Doron-Bargo) [inline logziosender.go:394]: few questions on this functions
1. where do you get the item value from ?
2. Its look like you are getting only 1 item - where is the batch?

**Review by Doron-Bargo — CHANGES_REQUESTED** [review]
Amazing work
I put some comments

File: logziosender.go [hunk 4]
```diff
@@ -203,25 +257,21 @@ func (l *LogzioSender) start() {
 func (l *LogzioSender) Stop() {
 	defer l.queue.Close()
 	l.Drain()
-
 }
 
 func (l *LogzioSender) tryToSendLogs() int {
 	resp, err := l.httpClient.Post(l.url, "text/plain", l.buf)
```
Comment (Doron-Bargo) [inline logziosender.go]: please send data compressed in Gzip 
also add the following headers
https://logzio.atlassian.net/browse/DEV-25393

**Review by Doron-Bargo — CHANGES_REQUESTED** [review]
please add gzip and the relevant headers

File: inMemoryQueue/inMemoryQueue.go (hunk 1 above)
Comment (yotamloe) [reply inMemoryQueue/inMemoryQueue.go]: We return this err so `dequeueUpToMaxBatchSize()` will know when to stop dequeuing

File: logziosender.go (hunk 3 above)
Comment (yotamloe) [reply logziosender.go:394]: 1 )`dequeueUpToMaxBatchSize()` this method loads items to the sender buffer and only returns the buffer size

2) The sender type has a buffer:
```go
type LogzioSender struct {
	queue             genericQueue
	drainDuration     time.Duration
	buf               *bytes.Buffer //here
	draining          atomic.Bool
	mux               sync.Mutex
	token             string
	url               string
	debug             io.Writer
	diskThreshold     float32
	checkDiskSpace    bool
	fullDisk          bool
	checkDiskDuration time.Duration
	dir               string
	httpClient        *http.Client
	httpTransport     *http.Transport
	// In memory Queue
	inMemoryQueue    bool
	inMemoryCapacity uint64
	logCountLimit    int
}
```

**Comment by yotamloe** [comment]
@Doron-Bargo I added the requested changes can you please review

**Review by Doron-Bargo — APPROVED** [review]
Amazing work

**Pull request merged**